Source/Scene/ModelExperimental/ModelExperimental.js

```diff
@@ -0,0 +1,369 @@
+   *
+   * @memberof ModelExperimental.prototype
+   *
+   * @type {Number}
+   * @default Axis.Y
+   * @readonly
+   *
+   * @private
+   */
+  upAxis: {
+    get: function () {
+      return this._sceneGraph._upAxis;
+    },
+  },
+
+  /**
+   * Gets the model's forward axis.
+   * By default, glTF 2.0 models are Z-forward according to the spec, however older
+   * glTF (1.0, 0.8) models used X-forward. Note that only Axis.X and Axis.Z are supported.
+   *
+   * @memberof ModelExperimental.prototype
+   *
+   * @type {Number}
+   * @default Axis.Z
+   * @readonly
+   *
+   * @private
+   */
+  forwardAxis: {
+    get: function () {
+      return this._sceneGraph._forwardAxis;
+    },
+  },
+
+  /**
+   * Gets the model's bounding sphere.
+   *
+   * @memberof ModelExperimental.prototype
+   *
+   * @type {BoundingSphere}
+   * @readonly
+   *
+   * @private
+   */
+  boundingSphere: {
+    get: function () {
+      //>>includeStart('debug', pragmas.debug);
+      if (!this._ready) {
+        throw new DeveloperError(
+          "The model is not loaded. Use ModelExperimental.readyPromise or wait for ModelExperimental.ready to be true."
+        );
+      }
+      //>>includeEnd('debug');
+
+      return this._sceneGraph._boundingSphere;
+    },
+  },
+
+  /**
+   * This property is for debugging only; it is not for production use nor is it optimized.
+   * <p>
+   * Draws the bounding sphere for each draw command in the model.  A glTF primitive corresponds
+   * to one draw command.  A glTF mesh has an array of primitives, often of length one.
+   * </p>
+   *
+   * @memberof ModelExperimental.prototype
+   *
+   * @type {Boolean}
+   *
+   * @default false
+   */
+  debugShowBoundingVolume: {
+    get: function () {
+      return this._debugShowBoundingVolume;
+    },
+    set: function (value) {
+      if (this._debugShowBoundingVolume !== value) {
+        this._debugShowBoundingVolumeDirty = true;
+      }
+      this._debugShowBoundingVolume = value;
+    },
+  },
+});
+
+/**
+ * Called when {@link Viewer} or {@link CesiumWidget} render the scene to
+ * get the draw commands needed to render this primitive.
+ * <p>
+ * Do not call this function directly.  This is documented just to
+ * list the exceptions that may be propagated when the scene is rendered:
+ * </p>
+ *
+ * @exception {RuntimeError} Failed to load external reference.
+ */
+ModelExperimental.prototype.update = function (frameState) {
+  if (!defined(this._defaultTexture)) {
+    this._defaultTexture = frameState.context.defaultTexture;
+  }
+
+  // Keep processing the glTF every frame until the main resources
+  // (buffer views) and textures (which may be loaded asynchronously)
+  // are processed.
+  if (!this._resourcesLoaded || !this._texturesLoaded) {
+    this._gltfLoader.process(frameState);
+  }
+
+  // short-circuit if the glTF resources aren't ready.
+  if (!this._resourcesLoaded) {
+    return;
+  }
+
+  if (!this._drawCommandsBuilt) {
```

Comment:
Am I understanding correctly that `buildDrawCommands` will only get called once on the first `update`?

Reply:
Correct, for now. In the future we do plan to have a mechanism to re-run the pipelines when settings change significantly. Though that's beyond the scope of this initial PR